cypress/integration/tests/rbac/custom-rules.test.ts

```diff
@@ -0,0 +1,185 @@
+
+import {
+    getRandomAnalysisData,
+    getRandomApplicationData,
+    hasToBeSkipped,
+    login,
+    logout,
+    preservecookies,
+} from "../../../utils/utils";
+import {
+    AnalysisStatuses,
+    application,
+    CredentialType,
+    UserCredentials,
+} from "../../types/constants";
+import { RulesRepositoryFields } from "../../types/types";
+import * as data from "../../../utils/data_utils";
+import { getRulesData } from "../../../utils/data_utils";
+import { CustomMigrationTarget } from "../../models/administration/custom-migration-targets/custom-migration-target";
+import { Analysis } from "../../models/migration/applicationinventory/analysis";
+import { UserArchitect } from "../../models/keycloak/users/userArchitect";
+import { UserMigrator } from "../../models/keycloak/users/userMigrator";
+import { CredentialsSourceControlUsername } from "../../models/administration/credentials/credentialsSourceControlUsername";
+import { User } from "../../models/keycloak/users/user";
+
+describe("Custom Rules RBAC operations", { tags: ["@tier2"] }, function () {
+    // Polarion TC 318
+
+    /**
+     *  There are three separated analysis to improve the test performance
+     *  This way once an application is created it doesn't need to be changed again for
+     *  migrator and architect.
+     */
+    let analysisWithPublicRules: Analysis;
+    let analysisWithPrivateRules: Analysis;
+    let analysisWithPrivateRulesNoCred: Analysis;
+
+    let sourceCredential: CredentialsSourceControlUsername;
+    let target: CustomMigrationTarget;
+    const architect = new UserArchitect(data.getRandomUserData());
+    const migrator = new UserMigrator(data.getRandomUserData());
+
+    before("Create test data", function () {
+        if (hasToBeSkipped("@tier2")) return;
+        User.loginKeycloakAdmin();
+        architect.create();
+        migrator.create();
+
+        login();
+
+        sourceCredential = new CredentialsSourceControlUsername(
+            data.getRandomCredentialsData(
+                CredentialType.sourceControl,
+                UserCredentials.usernamePassword,
+                true
+            )
+        );
+        sourceCredential.create();
+    });
+
+    beforeEach("Persist session", function () {
+        preservecookies();
+
+        cy.fixture("custom-rules").then(function (customRules) {
+            this.customRules = customRules;
+        });
+
+        cy.fixture("application").then(function (appData) {
+            this.appData = appData;
+        });
+
+        cy.fixture("analysis").then(function (analysisData) {
+            this.analysisData = analysisData;
+        });
+    });
+
+    it("Admin, Rules from public repository", function () {
+        analysisWithPublicRules = new Analysis(
+            getRandomApplicationData("bookServerApp", { sourceData: this.appData[0] }),
+            getRandomAnalysisData(this.analysisData[0])
+        );
+        analysisWithPublicRules.customRule = null;
```

Comment:
why do we need to set it to null ?
If it is an optional parameter and value not passed then it should be ignored in model.

Reply:
You're right, it is not needed. I forgot to remove it when I was doing some tests.